users/selfmanagedcredswithoutmfa_policy.tf

```diff
@@ -0,0 +1,149 @@
+# ------------------------------------------------------------------------------
+# IAM policy that allows users to administer their own user
+# accounts.  This policy is pretty much copied from here:
+# https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_examples_aws_my-sec-creds-self-manage.html
+#
+# The main difference is that we don't require these users to be
+# authenticated with MFA, since these accounts will be accessed
+# programatically (i.e. not via the AWS web console) where MFA is not
+# an option.
+#
+# Note that IAM policy variables (e.g. "aws:username") are used below. For
+# reference, refer to:
+# https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_variables.html
+# https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/iam_policy_document
+# ------------------------------------------------------------------------------
+
+data "aws_iam_policy_document" "self_managed_creds_without_mfa" {
+  # Allow users to view their own account information
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "iam:GetAccountPasswordPolicy",
+      "iam:GetAccountSummary",
+      "iam:ListVirtualMFADevices",
+    ]
+
+    resources = [
+      "*",
+    ]
+  }
+
+  # Allow users to administer their own passwords
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "iam:ChangePassword",
+      "iam:GetUser",
+    ]
+
+    resources = [
+      "arn:aws:iam::*:user/&{aws:username}",
+    ]
+  }
+
+  # Allow users to administer their own access keys
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "iam:CreateAccessKey",
+      "iam:DeleteAccessKey",
+      "iam:ListAccessKeys",
+      "iam:UpdateAccessKey",
+    ]
+
+    resources = [
+      "arn:aws:iam::*:user/&{aws:username}",
+    ]
+  }
+
+  # Allow users to administer their own signing certificates
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "iam:DeleteSigningCertificate",
+      "iam:ListSigningCertificates",
+      "iam:UpdateSigningCertificate",
+      "iam:UploadSigningCertificate",
+    ]
```

Comment:
Can you remind me what these certs are used for?

Reply:
I believe you can use a signing certificate as an alternative to the AWS programmatic credentials.

Reply:
I don't believe we currently use any signing certificates, but I'm not 100% sure.  As a reminder, the contents of this policy are borrowed from [this AWS example](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_examples_aws_my-sec-creds-self-manage.html) - this is mentioned in the comment at the top of our policy file:
https://github.com/cisagov/cool-accounts/blob/7073cdc641ab91adde6bff70aed1912eeb800475/users/selfmanagedcredswithoutmfa_policy.tf#L2-L4

Reply:
We don't use any signing certificates.